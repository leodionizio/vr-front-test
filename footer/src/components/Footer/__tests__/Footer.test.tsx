import { render, screen } from '@testing-library/react';
import { Footer } from "../";

describe("Footer component", () => {
  it("renders the VR Benefícios logo", () => {
    render(<Footer />);

    const logo = screen.getByAltText("VR Benefícios Logo");
    expect(logo).toBeInTheDocument();
  });

  it("renders the copyright text", () => {
    render(<Footer />);

    const copyrightText = screen.getByText(/todos os direitos reservados/i);
    expect(copyrightText).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Footer />);

    expect(asFragment()).toMatchSnapshot();
  });
});
