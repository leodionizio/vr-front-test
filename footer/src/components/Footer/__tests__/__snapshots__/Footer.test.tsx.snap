// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="sc-blHHSb dhcDNg"
  >
    <img
      alt="VR Benefícios Logo"
      class="sc-gtLWhw ahtgq"
      src="test-file-stub"
    />
    <p
      class="sc-egkSDF TpFTI"
    >
      © 2024 VR Benefícios - Todos os direitos reservados
    </p>
  </footer>
</DocumentFragment>
`;
